File: connection-backend/src/main/java/de/florianschmitt/rest/admin/DashboardController.java
package de.florianschmitt.rest.admin;

import org.springframework.beans.factory.annotation.Autowired;
import org.springframework.http.HttpStatus;
import org.springframework.http.MediaType;
import org.springframework.http.ResponseEntity;
import org.springframework.web.bind.annotation.RequestMapping;
import org.springframework.web.bind.annotation.RequestMethod;
import org.springframework.web.bind.annotation.RestController;

import de.florianschmitt.model.rest.EDashboardDTO;
import de.florianschmitt.service.DashboardService;

@RestController
@RequestMapping("/admin/dashboard")
class DashboardController {

    @Autowired
    private DashboardService service;

    @RequestMapping(path = "/info", method = RequestMethod.GET, produces = MediaType.APPLICATION_JSON_UTF8_VALUE)
    ResponseEntity<EDashboardDTO> get() {
        EDashboardDTO info = service.info();
        return new ResponseEntity<>(info, HttpStatus.OK);
    }
}

File: src/main/kotlin/de/florianschmitt/rest/admin/DashboardController.kt
package de.florianschmitt.rest.admin

import de.florianschmitt.model.rest.EDashboardDTO
import de.florianschmitt.service.DashboardService
import org.springframework.beans.factory.annotation.Autowired
import org.springframework.http.MediaType
import org.springframework.http.ResponseEntity
import org.springframework.web.bind.annotation.GetMapping
import org.springframework.web.bind.annotation.RequestMapping
import org.springframework.web.bind.annotation.RestController

@RestController
@RequestMapping("/admin/dashboard")
internal class DashboardController {

    @Autowired
    private lateinit var service: DashboardService

    @GetMapping(path = ["/info"], produces = [MediaType.APPLICATION_JSON_UTF8_VALUE])
    fun get(): ResponseEntity<EDashboardDTO> = ResponseEntity.ok(service.info())
}
